server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -141,15 +142,17 @@
     public Coordinator(String nodeName, Settings settings, ClusterSettings clusterSettings, TransportService transportService,
```

Comment:
I think we're at the point when the number of Coordinator constructor parameters are unmanageable. Can we possibly add a JavaDoc to the constructor describing how specific dependency is used by Coordinator?

Reply:
I know what you mean, but most of them are of very specific types. I added docs for the ones whose types don't make their meaning so clear in d9dbf2d (including the one added here).